### PR TITLE
FIX] fixes several plotting details in glm reports

### LINKED ISSUE #2405: Misc issues in Nistats reports
Here are some  issues I have encountered using nistats reports.
* Does not seem to take into account use-provided parameters for thresholding 
* the contrast maps only have the positive part displayed
* multi-dimensional contrasts do not display well.

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #2405

Changes proposed in this pull request:

- [x] only display FIR delay if FIR is used
- [x] make sure the run number does not overlap with the design matrix also only display it if there is more then one run
- [x] change number of significant figure in parameters for thresholding to make sure default alpha is properly displayed and use scientific notation for very small alphas
- [x] add "z-score" next to color bar
  - [x] handle black background
- [x] for `plot_type="slice"`
  - [x] use symmetric blue to white colormap when data has both positive and negative values
  - [x] use "red_transparent_full_alpha_range" colormap when data only has positive values
  - [x] use "blue_transparent_full_alpha_range" colormap when data only has negative values 
- [x] use a proper color map for glass brain

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Run number next to design matrices

image

**Comment by Remi-Gau** [comment]
Add unit to color bar

image

**Comment by Remi-Gau** [comment]
Note: consider doing the colormap modifications in a separate PR as the default choice may require a proper discussion.

**Comment by Remi-Gau** [comment]
Note: review and visualizing the outcome would be easier if https://github.com/nilearn/nilearn/issues/2194#issuecomment-1884889353 was addressed first

**Comment by Remi-Gau** [comment]
- blocked by #4267

File: nilearn/reporting/glm_reporter.py [hunk 1]
```diff
@@ -953,12 +961,30 @@ def _stat_map_to_svg(
         SVG Image Data URL representing a statistical map.
 
     """
+    data = safe_get_data(stat_img, ensure_finite=True)
+    stat_map_min = np.nanmin(data)
+    stat_map_max = np.nanmax(data)
+    print(stat_map_min)
```
Comment (bthirion) [inline nilearn/reporting/glm_reporter.py:967]: This is temporary, I guess ?

Comment (Remi-Gau) [reply nilearn/reporting/glm_reporter.py:967]: yup... 😰

**Comment by Remi-Gau** [comment]
Requested a full build to compare to the GLM reports before and after.

Will post the links once the builds are done.

**Comment by Remi-Gau** [comment]
To compare:

GLM reports before: https://nilearn.github.io/dev/modules/generated_reports/glm_reports_examples.html

GLM reports after: https://output.circle-artifacts.com/output/job/72355410-f50c-4f00-86ef-fa4c2b967fbc/artifacts/0/dev/modules/generated_reports/glm_reports_examples.html

**Review by bthirion — APPROVED** [review]
LGTM.

**Pull request merged**